hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/state/HdfsZkStore.java

```diff
@@ -38,9 +38,13 @@ public HdfsZkStore(HdfsFrameworkConfig hdfsFrameworkConfig) {
     return value;
   }
 
-  public void setRawValueForId(String id, byte[] frameworkId) throws ExecutionException, InterruptedException {
+  public void setRawValueForId(String id, byte[] newRawValue) throws ExecutionException, InterruptedException {
     Variable value = state.fetch(id).get();
-    value = value.mutate(frameworkId);
+    if (newRawValue == null) {
```

Comment:
Instead of setting the raw value for the ID to null, wouldn't it be preferable to remove it from the ZK State store instead? Then we could remove the null check here.